tests/system/src/test/java/test/javafx/embed/swing/JDK8220484Test.java

```diff
@@ -0,0 +1,155 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.javafx.embed.swing;
+
+import javafx.application.Platform;
+import javafx.embed.swing.JFXPanel;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import junit.framework.AssertionFailedError;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.lang.reflect.Field;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static test.util.Util.TIMEOUT;
+
+public class JDK8220484Test {
+    static CountDownLatch launchLatch;
+
+    private static MyApp myApp;
+
+    private static Timer t;
+    static int cnt;
+
+    @BeforeClass
+    public static void setupOnce() {
+        System.setProperty("sun.java2d.uiScale.enabled", "true");
+        System.setProperty("sun.java2d.uiScale", "125%");
+        System.setProperty("glass.win.uiScale", "1.25");
+        System.setProperty("glass.gtk.uiScale", "1.25");
+        launchLatch = new CountDownLatch(1);
+
+        // Start the Application
+        SwingUtilities.invokeLater(() -> {
+            myApp = new MyApp();
+        });
+
+        try {
+            if (!launchLatch.await(5 * TIMEOUT, TimeUnit.MILLISECONDS)) {
+                throw new AssertionFailedError("Timeout waiting for Application to launch (" + (5 * TIMEOUT) + " seconds)");
+            }
+        } catch (InterruptedException ex) {
+            AssertionFailedError err = new AssertionFailedError("Unexpected exception");
+            err.initCause(ex);
+            throw err;
+        }
+    }
+
+    @AfterClass
+    public static void teardownOnce() {
+        Platform.exit();
+    }
+
+    @Test
+    public void testScale() throws Exception {
+        // Get the Swing-side BackBuffer
+        Field fpixelsIm = JFXPanel.class.getDeclaredField("pixelsIm");
+        fpixelsIm.setAccessible(true);
+        BufferedImage pixelsIm = (BufferedImage) fpixelsIm.get(myApp.jfxPanel);
+
+
+        assertEquals(127, pixelsIm.getWidth());
+        assertEquals(127, pixelsIm.getHeight());
+
+        // if all is ok, this area has a gray shading
+        // if the buffer is off, there is a dark gray diagonal which should be the right border
+        Color c = new Color(181, 181, 181);
+        int colorOfDiagonal = c.getRGB();
+        for (int x = 10; x < 45; x++) {
+            for (int y = 90; y < 115; y++) {
+                if(colorOfDiagonal == pixelsIm.getRGB( x, y )) {
```

Comment:
Are you sure that an equality test will work on all platforms and configurations? We usually use a tolerance when comparing colors whose components are other than 0 or 255.

Somewhat related to this, is the expected value of `181` coming from the default value of the button border? It might be more robust to fill the Scene with a stroked rectangle whose color you control (e.g. set to black). Either that or set the color of the button border using an inline CSS style so you aren't dependent on the default inherited from the `modena.css` style sheet.

Minor: add a space after the `if` and remove the extra spaces surrounding the function arguments `x, y`.

Reply:
done, a rectangle with a defined background & border color. That is definitely more stable.